datalad/distribution/siblings.py

```diff
@@ -0,0 +1,382 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Plumbing command for managing sibling configuration"""
+
+__docformat__ = 'restructuredtext'
+
+
+import logging
+
+from os.path import basename
+from os.path import relpath
+
+# XXX confusing: we have urljoin, _urljoin, dlurljoin
+from datalad.distribution.add_sibling import _urljoin
+
+from datalad.interface.base import Interface
+from datalad.interface.utils import eval_results
+from datalad.interface.utils import build_doc
+from datalad.interface.results import get_status_dict
+from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureChoice
+from datalad.support.constraints import EnsureNone
+from datalad.support.param import Parameter
+from datalad.support.exceptions import CommandError
+from datalad.support.exceptions import InsufficientArgumentsError
+from datalad.support.network import RI
+from datalad.interface.common_opts import recursion_flag
+from datalad.interface.common_opts import recursion_limit
+from datalad.interface.common_opts import as_common_datasrc
+from datalad.interface.common_opts import publish_depends
+from datalad.interface.common_opts import publish_by_default
+from datalad.interface.common_opts import annex_wanted_opt
+from datalad.interface.common_opts import annex_group_opt
+from datalad.interface.common_opts import annex_groupwanted_opt
+from datalad.interface.common_opts import inherit_opt
+from datalad.distribution.dataset import require_dataset
+from datalad.utils import swallow_logs
+from datalad.dochelpers import exc_str
+
+from .dataset import EnsureDataset
+from .dataset import datasetmethod
+
+lgr = logging.getLogger('datalad.distribution.siblings')
+
+
+def _mangle_urls(url, ds_name):
+    if not url:
+        return url
+    return url.replace("%NAME", ds_name.replace("/", "-"))
+
+
+@build_doc
+class Siblings(Interface):
+    """
+
+    The following properties are reported (if possible) for each matching
+    subdataset record.
+
+    "name"
+        Name of the sibling
+
+    "path"
+        Absolute path of the dataset
+
+    """
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to configure.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the input and/or the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        name=Parameter(
+            args=('-s', '--name',),
+            metavar='NAME',
+            doc="""name of the sibling""",
```

Comment:
Add a sentence, pointing out that the hostname for the given URL is used as the default.